src/metabase/search/filter.clj

```diff
@@ -89,9 +89,7 @@
     [:or
      [:= :collection.personal_owner_id current-user-id]
      (into [:or]
-           (let [your-collection-ids (t2/select-pks-vec :model/Collection :personal_owner_id [:= current-user-id])
-                 child-patterns      (for [id your-collection-ids] (format "/%d/%%" id))]
-             (for [p child-patterns] [:like :collection.location p])))]
+           [:like :collection.location (format "/%d/%%" (t2/select-one-pk :model/Collection :personal_owner_id [:= current-user-id]))])]
```

Comment:
Is there a constraint in the db that prevents there being multiple personal collections?

Reply:
This is awesome then ❤️